contracts/stake/StakedTokenV3.sol

```diff
@@ -479,4 +575,41 @@ contract StakedTokenV3 is StakedTokenV2, IStakedTokenV3, RoleManager {
 
     emit Redeem(from, to, amountToRedeem, underlyingToRedeem);
   }
+
+  /**
+   * @dev searches a exchange Rate by block number. Uses binary search.
+   * @param blockNumber the block number being searched
+   **/
+  function _searchExchangeRateByBlockNumber(uint256 blockNumber) internal view returns (uint256) {
+    require(blockNumber <= block.number, 'INVALID_BLOCK_NUMBER');
+
+    if (_countExchangeRateSnapshots == 0) {
```

Comment:
When this condition can appear? I suppose the `initialize()` of this contract should snapshot the exchange rate, so there can't be a case where there is no count of snapshots when try to do binary search @dhadrien